packages/xstate-graph/src/graph.ts

```diff
@@ -342,7 +342,12 @@ export function getSimplePathsAsArray<
   return keys(result).map((key) => result[key]);
 }
 
-export function toDirectedGraph(stateNode: StateNode): DirectedGraphNode {
+export function toDirectedGraph(
+  machineNode: MachineNode | StateNode
+): DirectedGraphNode {
+  const stateNode =
+    machineNode instanceof StateNode ? machineNode : machineNode.root;
```

Comment:
In general, I like to avoid relying on `instanceof` checks for any code that escapes a boundary of a single package - this would fail with 2 copies of XState in `node_modules`.

I don't think that's something that should be fixed/handled as part of this PR though but perhaps, we should add a TODO comment here?

Reply:
```suggestion
    machineNode instanceof StateNode ? machineNode : machineNode.root; // TODO: accept only machines
```